src/Components/GEOSldas_GridComp/GEOSlandassim_GridComp/clsm_ensupd_upd_routines.F90

```diff
@@ -4430,13 +4488,177 @@ subroutine cat_enkf_increments(                               &
           end if
           
        end do
+       ! ----------------------------------
+
+     
+    case (11) select_update_type  ! 1d snow analysis (Toure et al. 2018 empirical gain); snow cover fraction obs
+       
+       if (logit) write (logunit, *) 'get 1d snow increments (Toure et al. 2018 empirical gain); snow cover fraction obs'
+       
+       ! make sure maximum SWE increment is less than WEMIN (adding more snow than WEMIN makes no sense)
+
+       if (max_incr_swe>WEMIN)  call ldas_abort(LDAS_GENERIC_ERROR, Iam, 'must use max_incr_swe<=WEMIN')
+       
+       !identify the species ID number of interest       
+       N_select_varnames  = 1      
+       select_varnames(1) = 'asnow'
+  
+       call get_select_species(                                           &
+            N_select_varnames, select_varnames(1:N_select_varnames),      &
+            N_obs_param, obs_param, N_select_species, select_species )
+       
+       allocate(select_tilenum(1))
+       
+       swe_incr = 0.   ! initialize to NO CHANGE
+
+       ! loop through tiles and compute increments
+       
+       do kk=1,N_catd
+          
+          ! find observations for tile kk
+          
+          select_tilenum(1) = l2f(kk)
+          
+          ! if(logit) write(logunit,*) 'kk:', kk
+          
+          call get_ind_obs(                                           &
+               N_obs,            Observations,                        &
+               1,                select_tilenum,                      &
+               N_select_species, select_species(1:N_select_species),  &
+               N_selected_obs,   ind_obs )
+          
+          if (N_selected_obs == 1) then  ! COULD THERE EVER BE MORE THAN ONE OBS HERE??? TBD reichle 20221007 
+             
+             do n_e=1,N_ens ! compute analysis separately for each ensemble member
+                
+                ! 1. Calculate model forecast snow area and SWE
+
+                asnow_fcst  = asnow(kk, n_e)
+                swe_fcst    = sum(cat_progn(kk, n_e)%wesn(1:N_snow))
+                
+                ! 2. Calculate SWE increment based on Toure et al (2018) eq 1
+
+                if     (asnow_fcst .lt. Observations(ind_obs(1))%obs * alpha_threshold) then
+                   
+                   ! ADD SNOW: SCA of model is less than observed SCA
+                   
+                   swe_incr(kk,n_e)                                                                     &
+                        = max_incr_swe * (Observations(ind_obs(1))%obs - asnow_fcst/alpha_threshold)
+                   
+                elseif (Observations(ind_obs(1))%obs .lt. beta_threshold) then
+                   
+                   ! REMOVE SNOW: IF SCA of model is greater than observed SCA
+                   
+                   swe_incr(kk,n_e) = (-1.) *  max_incr_swe * (1. - Observations(ind_obs(1))%obs/beta_threshold)
```

Comment:
After thinking about this overnight I realized that the "remove snow" equation should read as follows:
`swe_incr(kk,n_e) = (-1.) * swe_fcst * (1. - Observations(ind_obs(1))%obs/beta_threshold)`
That is, replace `max_incr_swe` with `swe_fcst`, which makes the increment should be proportional to `swe_fcst`.
After realizing this, I then checked the first draft and the final published version of Toure et al 2018, both of which also use `swe_fcst` (_SWE_m_, which is never defined...) and not `max_incr_swe`.
Note that because `asnow_obs<beta` per the elseif condition, we have `-swe_fcst <= swe_incr < 0`.
Does this make sense? 
Also, it's not clear to me immediately what the value of `beta` should be.  This requires more thought and (probably) data.
cc: @lcandre2 @gmao-qliu @amfox37

Reply:
Thanks @gmao-rreichle. 

Replacing max_incr_swe makes sense. I know we went back and forth on this previously because using max_incr_swe is what is needed to reproduce Figure 2 and it seemed odd that we could remove more snow than could be added (since max_incr_swe remains in the first if statement). But, a higher rate of removal during assimilation makes sense if MOD10C1 overestimates snow presence. From a quick lit search, this seems to be the case, at least for North America. Globally, this isn't so clear, so I'll look into it more.

Beta is (unclearly) related to the difference between the model and MODIS SCF, so I'll come up with a clearer way to define this and other parameters since they will need to be considered when switching satellite platforms. 

@gmao-rreichle @gmao-qliu @amfox37

Reply:
@amfox37 : The large negative increments make sense because a zero snow cover obs should be reasonably reliable.  If MODIS doesn't see snow, there shouldn't be any in the model.  Whether we should use WEMIN (13 mm nowadays) or SWE_fcst is up for discussion.  The graphic in Toure et al does not really show increments for SWE_fcst>WEMIN.  In case SWE_fcst>WEMIN and SCF_obs=0, we might as well remove all of the snow rather than just the first WEMIN kg/m2.
Re. the second part of the conditional:  I think I removed this in my attempts to simplify the implementation because the opening if condition already ensures that SCF_fcst > SCF_obs*alpha if we get into the elseif statement.  That is, the second part of the conditional should be redundant.  Or did I get this wrong?

Reply:
After revisiting Toure et al., I am convinced that max_swe_incr is used when Modis obs are less than beta. F2 has an increment range of -5 to 5 the DA-OL figures do not visually display a difference beyond 5 mmwe (even though the color bar goes to 6 and in the sensitivity analysis, there is the following statement: " ... it was concluded that max_swe_incr = 5 mm is a reasonable approximation for the maximum amount of snow that can be added to (**or removed from**) a catchment during the ... rule-based update procedure". But, now I think that what is plotted/used in Toure et al is option 2b below. 

max_swe_incr cannot be used alone during snow removal -- it needs to be scaled with the fcstSCF on line 4552. And honestly, I don't know why it isn't in the code, probably a mistake on my part. This would also pertain to using swe_min.

I think there are two, possibly 3 options: 
 1. swe_incr = (-1.0) * (fcstSWE) * (1 - obsSCF / beta)
This scales the increment by the model SWE, which produces lower negative increments when the SCF is below 1, but can produce really high negative increments if there is full, very deep snow cover.

**2a.** swe_incr = (-1.0) * (swe_min * fcstSCF ) * (1 - obsSCF / beta)
In this case, I've included the forecast SCF. In the first option, the results are the same at SCF>1, but when fcstSCF = 1 and ObsSCF = 0, the maximum negative increment would be capped at swe_min.

OR
**2b.** swe_incr = (-1.0) * (max_swe_incr * fcstSCF ) * (1 - obsSCF / beta)
Here, we could use max_swe_incr instead of min_swe. This would essentially lighten the touch of the negative increment. But, it would probably be best to determine the value of max_swe_incr with a sensitivity study.

Finally, BETA. 40% was chosen because the difference between the model and obs SCF rarely exceeds 40%. I am actually looking into this and coming up with a more precise way to determine this. Same with ALPHA.